docx/text/paragraph.py

```diff
@@ -39,6 +41,77 @@ def add_run(self, text=None, style=None):
             run.style = style
         return run
 
+    def add_field(self, instrText=None):
+        """
+        Adds new field `w:fldChar` to run. Pass `instrText` to specify
+        filed instruction.
+        """
+        self.add_run().add_fldChar()
+        if instrText:
+            self.add_run().add_instrText(instrText)
+        self.add_run().add_fldChar(fldCharType='end')
+
+    def add_sdt(self, tag_name, text='', alias_name='', temporary='false', locked='unlocked',
+                placeholder_txt=None, style='Normal', bold=False, italic=False, type=SdtType.RICH_TEXT):
+        """
+        Adds new Structured Document Type ``w:sdt`` field to the Paragraph element.
+        """
+
+        def apply_run_formatting(rPr, style='Normal', bold=False, italic=False, underline=False):
+            if style != 'Normal':
+                rStyle = rPr._add_rStyle()
+                rStyle.set('{%s}val' % nsmap['w'], style)
+            if bold:
+                rPr._add_b()
+                rPr._add_bCs()
+            if italic:
+                rPr._add_i()
+            # TODO: impl underline
+
+        sdt = self._p._new_sdt()
+
+        sdtPr = sdt._add_sdtPr()
+        alias_name = alias_name or tag_name
+
+        # set styling on sdt lvl
+        rPr = sdtPr.get_or_add_rPr()
+        apply_run_formatting(rPr, style, bold, italic)
+
+        tag = sdtPr._add_tag_name()
+        tag.set('{%s}val' % nsmap['w'], tag_name)
+
+        alias = sdtPr._add_alias()
+        alias.set('{%s}val' % nsmap['w'], alias_name)
+
+        temp = sdtPr._add_temporary()
+        temp.set('{%s}val' % nsmap['w'], temporary)
+
+        sdtContent = sdt._add_sdtContent()
+
+        if not text:
+            r = sdtContent._add_r()
```

Comment:
Would it make sense to move this line before if?

Reply:
yes it repeats itself